src/jdk.hotspot.agent/macosx/native/libsaproc/MacosxDebuggerLocal.m

```diff
@@ -160,16 +160,23 @@ static void throw_new_debugger_exception(JNIEnv* env, const char* errMsg) {
   return (struct ps_prochandle*)(intptr_t)ptr;
 }
 
-#if defined(amd64)
+#if defined(__i386__)
+    #define hsdb_thread_state_t     x86_thread_state32_t
```

Comment:
It doesn't support i386, and it reverts some of the review left on #2 .

Reply:
some leftovers